pkg/api/v1/pod/util.go

```diff
@@ -63,21 +63,30 @@ func FindPort(pod *v1.Pod, svcPort *v1.ServicePort) (int, error) {
 
 // TODO: remove this function when init containers becomes a stable feature
 func SetInitContainersAndStatuses(pod *v1.Pod) error {
-	var initContainersAnnotation string
-	initContainersAnnotation = pod.Annotations[v1.PodInitContainersAnnotationKey]
-	initContainersAnnotation = pod.Annotations[v1.PodInitContainersBetaAnnotationKey]
-	if len(initContainersAnnotation) > 0 {
+	// If there is a beta annotation, copy to alpha key.
+	// See commit log for PR #31026 for why we do this.
+	if valueBeta, okBeta := pod.Annotations[v1.PodInitContainersBetaAnnotationKey]; okBeta {
+		pod.Annotations[v1.PodInitContainersAnnotationKey] = valueBeta
```

Comment:
> when a kubelet updates, it definitely needs to look at beta status annotation, but it can ignore beta spec annotation (because api server is required to be newer)

That'll not hurt anything if we do so, right? I'd like to re-use this func for `convert.go` as mentioned in #41004 .